client/src/utils/ajax.ts

```diff
@@ -51,16 +55,93 @@ async function request<T>(
 /** GET **/
 
 interface SessionUser {
-  user: UserType;
+  user: { [username: string]: UserType };
   sessionMeta: { activeDonations: number };
-  result: string;
 }
-export function getSessionUser(): Promise<SessionUser> {
-  return get('/user/get-session-user');
+
+type challengeFilesForFiles = {
+  files: Array<Omit<ChallengeFile, 'fileKey'> & { key: string }>;
+} & Omit<CompletedChallenge, 'challengeFiles'>;
+
+type ApiSessionResponse = Omit<SessionUser, 'user'>;
+type ApiUser = {
+  user: {
+    [username: string]: ApiUserType;
+  };
+  result: string;
+};
+
+type ApiUserType = Omit<UserType, 'completedChallenges'> & {
+  completedChallenges: challengeFilesForFiles[];
```

Comment:
```suggestion
  completedChallenges?: challengeFilesForFiles[];
```
Not all user objects have completedChallenges, it seems.

Reply:
Where are you seeing/getting this? How does that come about?

Reply:
Since timelines can be private, the api doesn't always return this property.  See: https://github.com/freeCodeCamp/freeCodeCamp/blob/895987007f203379f5a0829311c8c45d55959052/api-server/src/common/models/user.js#L802-L812

Reply:
But, what I am seeing is that it is always returned. It is just sometimes an empty array?

Reply:
Sorry, I saw that completedChallenges could be undefined and just skimmed this code. It's the other return above.